inst/tests/test-select_rows.r

```diff
@@ -32,3 +32,18 @@ test_that("it correctly selects rows by function", {
   expect_equal(dd, dd2[3, ]) 
 })
 
+test_that("it correctly selects rows by function and cols_to_check with whole = FALSE", {
+  dd <- data.frame(x = c(1,2,3), y = c(4,5,6), z = c(-1,-2,999))
+  dd2 <- dd
+  select_rows(dd, function(x) {sum(x) == 0}, whole = FALSE, cols_to_check = c('x', 'z'))
```

Comment:
@tonglu for the `whole = FALSE` case, we are it will use the `apply` function over the dataframe. Hence `x` is a row of the data frame. In this case, I am only checking the columns `x` and `y`. You can see by construction of `dd` the row wise sum for row1 and row2 is 0 but not row3.